Shuttle.Core.Pipelines/ReusableObjectPool.cs

```diff
@@ -28,23 +28,20 @@ public TReusableObject Get(Type key)
 
             lock (Lock)
             {
-                if (!_pool.ContainsKey(key))
+                if (!_pool.TryGetValue(key, out var reusableObjects))
                 {
-                    _pool.Add(key, new List<TReusableObject>());
+                    reusableObjects = new List<TReusableObject>();
+                    _pool.Add(key, reusableObjects);
                 }
 
-                if (_pool.Count > 0)
+                if (reusableObjects.Count > 0)
                 {
-                    var reusableObjects = _pool[key];
+                    int lastIndex = reusableObjects.Count - 1;
```

Comment:
It does not matter that we get the first or the last, right? Removing the last is much faster than removing the first item from a list.